src/sqlfluff/cli/formatters.py

```diff
@@ -55,16 +55,16 @@ def format_violation(violation, max_line_length=90):
         if violation.line_no is not None:
             line_elem = f"{violation.line_no:4d}"
         else:
-            line_elem = "   -"
+            line_elem = "   -"  # pragma: no cover
         if violation.line_pos is not None:
             pos_elem = f"{violation.line_pos:4d}"
         else:
-            pos_elem = "   -"
-    else:
+            pos_elem = "   -"  # pragma: no cover
+    else:  # pragma: no cover
         raise ValueError(f"Unexpected violation format: {violation}")
```

Comment:
Noting that this function requires `violation` is a `SQLBaseError`, would you mind changing the definition line to:
```
def format_violation(violation: SQLBaseError, max_line_length: int=90):
```

I like type hints. Makes my editor smarter than I am. 😉

Reply:
But is that always true? The current function definition is below so it does that check:

```py
def format_violation(violation, max_line_length=90):
    """Format a violation."""
    if isinstance(violation, SQLBaseError):
        ...
    else:
        raise ValueError(f"Unexpected violation format: {violation}")
```

So by adding that type in the definition we can never hit that final else? Which maybe we shouldn't ever really hit, but which we currently cater for (to a point).

Reply:
Note that type hints don't enforce any behavior at runtime, just "build time" (mypy). This is just a nitpick, no big deal.